dev/source-release.sh

```diff
@@ -21,12 +21,12 @@
 set -e
 
 if [ -z "$1" ]; then
-  echo "Usage: $0 <version> <rc-num>"
+  echo "Usage: $0 <version-num> <rc-num>"
```

Comment:
I added `version-num`, as when running the release script, I occasionally found myself passing in `./dev/source-release.sh apache-iceberg-0.10.1 rc2`. In the future, we might consider opening an issue to detect whether or not the passed in $1 and $2 are valid floats, but that's beyond the scope of this PR.